plugins/qeta/src/components/MarkdownEditor/MarkdownEditor.tsx

```diff
@@ -1,23 +1,104 @@
+/* eslint-disable no-console */
 import React from 'react';
 import ReactMde from 'react-mde';
+
+import { Config } from '@backstage/config';
 import { MarkdownContent } from '@backstage/core-components';
 import 'react-mde/lib/styles/css/react-mde.css';
 import 'react-mde/lib/styles/css/react-mde-editor.css';
 import 'react-mde/lib/styles/css/react-mde-toolbar.css';
 import { useStyles } from '../../utils/hooks';
+import FileType from 'file-type';
+import { ErrorApi, errorApiRef, useApi } from '@backstage/core-plugin-api';
+
+const DEFAULT_IMAGE_SIZE_LIMIT = 2500000;
+const SUPPORTED_FILES_TYPES = [
+  'image/png',
+  'image/jpg',
+  'image/jpeg',
+  'image/gif',
+];
+
+export const submitImage = async (
+  config: Config,
+  file: Blob,
+  erroAlert: ErrorApi,
+) => {
+  const backendBaseUrl = config.getString('backend.baseUrl');
+  const qetaUrl = `${backendBaseUrl}/api/qeta/attachments`;
+  const formData = new FormData();
+
+  formData.append('image', file);
```

Comment:
Maybe add question/answer id to the data? This would allow to create relation in database so that when question or answer is deleted, it would automatically delete the attachments as well if they are stored in database. For attachments in filesystem.. Not sure how to clean those..

Reply:
That would be great, but I don't know how I would add it since the upload is done before the question is created.

In the cases of answers and editing, we were probably able to add.

For the case of the filesystem, we saved the path information locally in the database, we can use this information to delete it locally in a specific way.

Reply:
Oh true. So when posting the question, it should actually post the attachment IDs in that data.